src/map/battlefield.cpp

```diff
@@ -542,8 +542,11 @@ void CBattlefield::cleanupDynamis() {
             uint32 mobid = Sql_GetUIntData(SqlHandle, 0);
             CMobEntity* PMob = (CMobEntity*)zoneutils::GetEntity(mobid, TYPE_MOB);
 
-            if (PMob != nullptr)
-                PMob->PAI->Despawn();
+			// force mobs to the fully despawned state or else they won't be able to spawn on next Dynamis entry.
+			if (PMob != nullptr) {
+				PMob->FadeOut();
+				PMob->PAI->Internal_Respawn(0s);
```

Comment:
internal_respawn is called internal because nothing should ever be calling it from outside the AI - this can't stay, but we can work out the actual problem in irc if you want to chat with me when I'm at home